src/conn.rs

```diff
@@ -762,23 +761,24 @@ impl<const N: usize, P: ConnectionPeer> Connection<N, P> {
 
         match packet.packet_type() {
             PacketType::Syn => self.on_syn(packet.seq_num()),
-            PacketType::State => {
-                let delay = Duration::from_micros(packet.ts_diff_micros().into());
-                self.on_state(
-                    packet.seq_num(),
-                    packet.ack_num(),
-                    packet.selective_ack(),
-                    delay,
-                    now,
-                );
-            }
+            PacketType::State => self.on_state(packet.seq_num(), packet.ack_num()),
             PacketType::Data => self.on_data(packet.seq_num(), packet.payload()),
-            PacketType::Fin => {
-                self.on_fin(packet.seq_num(), packet.payload());
-            }
-            PacketType::Reset => {
-                self.on_reset();
+            PacketType::Fin => self.on_fin(packet.seq_num(), packet.payload()),
+            PacketType::Reset => self.on_reset(),
+        }
+
+        // Mark sent packets as acked, using the received ack_num
+        match packet.packet_type() {
+            PacketType::Syn => {}
+            PacketType::State | PacketType::Data | PacketType::Fin => {
+                let delay = Duration::from_micros(packet.ts_diff_micros().into());
+                if let Err(err) =
+                    self.received_ack(packet.ack_num(), packet.selective_ack(), delay, now)
+                {
+                    tracing::warn!(%err, ?packet, "ack does not correspond to known seq_num");
```

Comment:
we probably don't want to log the entire packet here

Reply:
Yeah, the debug is custom, and already skips the fields that are super wordy. It's been helpful once already!